ride-share.rb

```diff
@@ -0,0 +1,105 @@
+ride_share = {
+  DR0001: [
+    {
+      date: "3rd Feb 2016", cost: 10, rider_id: "RD0022", rating: 5,
+    },
+    {
+      date: "3rd Feb 2016", cost: 30, rider_id: "RD00015", rating: 4,
+    },
+    {
+      date: "5th Feb 2016", cost: 45, rider_id: "RD0003", rating: 2,
+    },
+  ],
+  DR0002: [
+    {
+      date: "3rd Feb 2016", cost: 25, rider_id: "RD0073", rating: 5,
+    },
+    {
+      date: "4th Feb 2016", cost: 15, rider_id: "RD0013", rating: 1,
+    },
+    {
+      date: "5th Feb 2016", cost: 35, rider_id: "RD0066", rating: 3,
+    },
+  ],
+  DR0003: [
+    {
+      date: "4th Feb 2016", cost: 5, rider_id: "RD0066", rating: 5,
+    },
+    {
+      date: "5th Feb 2016", cost: 50, rider_id: "RD0003", rating: 2,
+    },
+  ],
+  DR0004: [
+    {
+      date: "3rd Feb 2016", cost: 5, rider_id: "RD0022", rating: 5,
+    },
+    {
+      date: "4th Feb 2016", cost: 10, rider_id: "RD0022", rating: 4,
+    },
+    {
+      date: "5th Feb 2016", cost: 20, rider_id: "RD0073", rating: 5,
+    },
+  ],
+}
+
+#1. How many Rides has each driver given?
+puts "Driver amount of rides"
+ride_share.each do |driver, rides|
+  print "#{driver}: "
+  puts rides.count
+end
+
+#2. Total amount of money each driver has made
+puts "Total amount earned by each driver"
+ride_share.each do |driver, rides|
+  money = rides.sum do |record|
+    record[:cost]
+  end
+  puts "#{driver}: $#{money}"
+end
+
+#3. # - the average rating for each driver
+ride_share.each do |driver, rides|
+  average_rating = rides.sum do |record|
+    ((record[:rating].to_f) / rides.length).round(2)
+  end
+  puts "The average rating of #{driver} is #{average_rating}"
+end
+
+#4.- Which driver made the most money?
+highest_earnings = ride_share.map do |driver, rides|
+  { name: driver, total: rides.sum do |record|
+    record[:cost]
+  end }
+end
+
+richest_driver = highest_earnings.max_by do |driver|
+  driver[:total]
+end
+
+puts "The driver who made the most money: #{richest_driver[:name]}."
+
+#5. # - Which driver has the highest average rating? (Just like previous one)
+all_averages = ride_share.map do |driver, rides|
+  { name: driver, average: rides.sum do |record|
+    ((record[:rating].to_f) / rides.length).round(2)
+  end }
```

Comment:
This is mostly easy to read, but be aware, the `{` can be on it's own line or with the thing it's attached to. usually you'd see this formatted:

``` ruby
all_averages = ride_share.map do |driver, rides|{ 
  name: driver, average: rides.sum do |record|
    ((record[:rating].to_f) / rides.length).round(2)
  end 
}end
```